### PR TITLE
Category issue branch

### PR DESCRIPTION
**Pull request by BeratARPA**
The necessary steps for the relevant issue have been completed. The only problem here is that the "AppUserId" column is empty in the categories that need to be added while registering.

### CODE REVIEW

File: prExplorer/API/Controllers/AccountController.cs [hunk 1]
```diff
@@ -58,7 +58,38 @@ public async Task<ActionResult<UserDto>> Register(RegisterDto registerDto)
             {
                 DisplayName = registerDto.DisplayName,
                 UserName = registerDto.Username,
-                Email = registerDto.Email
+                Email = registerDto.Email,
+                Categories = new List<Category>
```
Comment (dsibinski) [inline prExplorer/API/Controllers/AccountController.cs]: I suggest storing the default categories somewhere in the database and only retrieving them here

File: prExplorer/Application/Categories/Delete.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                _context = context;
+            }
+
+            public async Task<Result<Unit>> Handle(Command request, CancellationToken cancellationToken)
+            {
+                var user = await _context.Users.FirstOrDefaultAsync(x => x.Id == _userAccessor.GetUserId());
+
+                if (user == null)
+                {
+                    // Eğer kullanıcı bulunamazsa hata işleme.
```
Comment (dsibinski) [inline prExplorer/Application/Categories/Delete.cs]: I suggest not using Turkish comments/texts in code

File: prExplorer/ApplicationTests/ProductTest/ListTest.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+            var mockMapper = new MapperConfiguration(cfg => { cfg.AddProfile(new MappingProfiles()); });
+            var mapper = mockMapper.CreateMapper();
+            var handler = new GetAllProductsTest.Handler(context, mapper, userAccessor);
+
+            #endregion
+
+            #region Assert
+
+            var value = handler.Handle(new GetAllProductsTest.Query(), CancellationToken.None).Result.Value;
```
Comment (dsibinski) [inline prExplorer/ApplicationTests/ProductTest/ListTest.cs]: As suggested in the other PR, this `GetAllProductsTest` should not exist. The test should test the real, "production" handlers

File: prExplorer/client-app/src/app/stores/categoryStore.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  loadCategories = async () => {
+    this.setLoadingInitial(true);
+    try {
+      const categories = await agent.Categories.list();
+      categories.forEach((category) => {
+        this.setCategory(category);
+      });
+      this.setLoadingInitial(false);
+    } catch (error) {
+      console.log(error);
```
Comment (dsibinski) [inline prExplorer/client-app/src/app/stores/categoryStore.ts:37]: `console.log` shouldn't be checked-in

File: prExplorer/client-app/src/app/stores/categoryStore.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    this.loading = true;
+    try {
+      await agent.Categories.delete(id);
+      runInAction(() => {
+        this.categoryRegistry.delete(id);
+        this.loading = false;
+        toast.success("Succesfully Deleted Category");
+      });
+    } catch (error) {
+      console.log(error);
```
Comment (dsibinski) [inline prExplorer/client-app/src/app/stores/categoryStore.ts:123]: why there still isn't proper and common errors handling implemented here?

**Review by dsibinski — APPROVED** [review]
Few remarks

File: prExplorer/API/Controllers/AccountController.cs (hunk 1 above)
Comment (aliyorulmazdev) [reply prExplorer/API/Controllers/AccountController.cs]: we can crate dummyUser and copy categories from that.

File: prExplorer/Application/Categories/Delete.cs (hunk 2 above)
Comment (aliyorulmazdev) [reply prExplorer/Application/Categories/Delete.cs]: solved latest commit

File: prExplorer/client-app/src/app/stores/categoryStore.ts (hunk 4 above)
Comment (aliyorulmazdev) [reply prExplorer/client-app/src/app/stores/categoryStore.ts:37]: We'll create new controller on API side for error logs.(Front-Back). Use seperate db (mongoDb cloud) there.

File: prExplorer/client-app/src/app/stores/categoryStore.ts (hunk 5 above)
Comment (aliyorulmazdev) [reply prExplorer/client-app/src/app/stores/categoryStore.ts:123]: We'll create new controller on API side for error logs.(Front-Back). Use seperate db (mongoDb cloud) there.

**Pull request merged**